### PR TITLE
Fixup py opencv pinning

### PR DESCRIPTION
**Pull request by hmaarrfk**
Checklist
* [ ] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by hmaarrfk** [comment]
With the following build string, one gets the 3 packages:

```
BUILD START: ['libopencv-4.1.2-py38_2.tar.bz2', 'py-opencv-4.1.2-py38h5ca1d4c_2.tar.bz2', 'opencv-4.1.2-py38_2.tar.bz2']
```

each unique to a specific version of python.

**Comment by hmaarrfk** [comment]
The important one to unique to a version of python is libopencv, which was specific to python 2.7

File: recipe/meta.yaml [hunk 1]
```diff
@@ -20,6 +20,9 @@
 # By putting all the generated files in 1 package, this makes the build process
 # much easier, at the expense of a few MBs in the 'lib' package.
 {% set version = "4.1.2" %}
+{% set PY_VER_MAJOR = PY_VER[0] %}
+{% set PY_VER_MINOR = PY_VER[2] %}
+
```
Comment (hmaarrfk) [inline recipe/meta.yaml]:
```suggestion
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -35,7 +37,10 @@ source:
     folder: opencv_contrib
 
 build:
-  number: 1
+  # Trailing 2 is the build number
```
Comment (hmaarrfk) [inline recipe/meta.yaml]:
```suggestion
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -35,7 +37,10 @@ source:
...
+  number: 2
+  # How to remove the dot in the py_ver???
```
Comment (hmaarrfk) [inline recipe/meta.yaml]:
```suggestion
```

**Comment by hmaarrfk** [comment]
```
anaconda upload /home/conda/feedstock_root/build_artifacts/linux-64/libopencv-4.1.2-py38_2.tar.bz2
anaconda upload /home/conda/feedstock_root/build_artifacts/linux-64/py-opencv-4.1.2-py38h5ca1d4c_2.tar.bz2
anaconda upload /home/conda/feedstock_root/build_artifacts/linux-64/opencv-4.1.2-py38_2.tar.bz2
```

```
anaconda upload /home/conda/feedstock_root/build_artifacts/linux-64/libopencv-4.1.2-py37_2.tar.bz2
anaconda upload /home/conda/feedstock_root/build_artifacts/linux-64/py-opencv-4.1.2-py37h5ca1d4c_2.tar.bz2
anaconda upload /home/conda/feedstock_root/build_artifacts/linux-64/opencv-4.1.2-py37_2.tar.bz2
```

I think this makes things better right?

@ocefpaf do merge if you are OK with this.

**Comment by ocefpaf** [comment]
I'm no expert but let's give this a go. Thanks!

**Comment by hmaarrfk** [comment]
I guess it will take about 1 hour to build.

**Pull request merged**